### PR TITLE
changed getter and setter for constraints to return sets.

### PR DESCRIPTION
**Pull request by shuangzhou**
This request provides update to constraints api as constraints selected need to be in a unique set as per dscs-227.

### CODE REVIEW

**Comment by shuangzhou** [comment]
@scoen could you please review this change?  Thanks.

**Comment by terenceingram** [comment]
As lists are ordered and the sets are not ordered, I wonder if that will make a difference somewhere in the UI or somewhere else?

**Comment by shuangzhou** [comment]
Good point, the calling app should use the LinkedHashSet to construct the multi-valued constraint.  I'll update the getConstraint to retain order.

**Comment by terenceingram** [comment]
I wonder how freemarker handles Sets? I wonder if you access it the same way as a list in freemarker.

**Comment by shuangzhou** [comment]
Looks like you can access it the same in freemarker.

File: src/amberdb/model/Work.java [hunk 1]
```diff
@@ -1270,12 +1272,17 @@ public IsPartOf getParentEdge() {
...
 
         @Override
-        public List<String> getConstraint() throws JsonParseException, JsonMappingException, IOException {          
-            return deserialiseJSONString(getJSONConstraint());
+        public Set<String> getConstraint() throws JsonParseException, JsonMappingException, IOException { 
+            List<String> list = deserialiseJSONString(getJSONConstraint());
+            LinkedHashSet<String> constraint = new LinkedHashSet<>();
+            for (String entry : list) {
+                constraint.add(entry);
+            }
```
Comment (scoen) [inline src/amberdb/model/Work.java]: can you use addAll here to reduce the lines of code ?

**Comment by shuangzhou** [comment]
It's updated.

**Comment by terenceingram** [comment]
@shuangzhou @scoen Shuang, I forgot to add that you may need to test the indexing code in Jelly. As you have changed the return type, I am not sure what the code that is indexing is expecting when it gets the constraint. It would definitely iterate over the list to add it as a multivalued field.

**Comment by shuangzhou** [comment]
@terenceingram I'll update Jelly with this change.  Thanks for the suggestion.

**Pull request merged**